tests/s3_data_plane_tests.c

```diff
@@ -8114,3 +8114,52 @@ static int s_test_s3_default_get_without_content_length(struct aws_allocator *al
 
     return AWS_OP_SUCCESS;
 }
+
+struct aws_string *test_helper(struct aws_allocator *allocator) {
```

Comment:
The function name `test_helper` is too generic and doesn't clearly communicate its purpose. Consider renaming to something more specific like `s_test_s3_build_endpoint_string` to match the naming convention of other helper functions in this codebase.